napari/utils/transforms/transform_utils.py

```diff
@@ -29,51 +29,220 @@ def compose_linear_matrix(rotate, scale, shear) -> np.array:
+        pitch, roll convention. Otherwise assume an nD rotation. Angles are
+        assumed to be in degrees. They can be converted from radians with
+        np.degrees if needed.
+    ndim : int
+        The desired dimensionality of the output transform component.
+
+    Returns
+    -------
+    np.ndarray
+        The rotate component as a 2D numpy array of size ndim.
+    """
+    full_rotate_mat = np.eye(ndim)
+    if rotate is not None:
+        rotate_mat = _make_rotate_mat(rotate)
+        rotate_mat_ndim = rotate_mat.shape[0]
+        full_rotate_mat[-rotate_mat_ndim:, -rotate_mat_ndim:] = rotate_mat
+    return full_rotate_mat
+
+
+def _make_rotate_mat(rotate):
     if np.isscalar(rotate):
-        # If a scalar is passed assume it is a single rotation angle
-        # for a 2D rotation
-        theta = np.deg2rad(rotate)
-        rotate_mat = np.array(
-            [[np.cos(theta), -np.sin(theta)], [np.sin(theta), np.cos(theta)]]
-        )
+        return _make_2d_rotation(rotate)
     elif np.array(rotate).ndim == 1 and len(rotate) == 3:
-        # If a 3-tuple is passed assume it is three rotation angles for
-        # a roll, pitch, and yaw for a 3D rotation. For more details see
-        # https://en.wikipedia.org/wiki/Rotation_matrix#General_rotations
-        alpha = np.deg2rad(rotate[0])
-        beta = np.deg2rad(rotate[1])
-        gamma = np.deg2rad(rotate[2])
-        R_alpha = np.array(
-            [
-                [np.cos(alpha), np.sin(alpha), 0],
-                [-np.sin(alpha), np.cos(alpha), 0],
-                [0, 0, 1],
-            ]
-        )
-        R_beta = np.array(
-            [
-                [np.cos(beta), 0, np.sin(beta)],
-                [0, 1, 0],
-                [-np.sin(beta), 0, np.cos(beta)],
-            ]
-        )
-        R_gamma = np.array(
-            [
-                [1, 0, 0],
-                [0, np.cos(gamma), -np.sin(gamma)],
-                [0, np.sin(gamma), np.cos(gamma)],
-            ]
-        )
-        rotate_mat = R_alpha @ R_beta @ R_gamma
-    else:
-        # Otherwise assume a full nD rotation matrix has been passed
-        rotate_mat = np.array(rotate)
-    n_rotate = rotate_mat.shape[0]
+        return _make_3d_rotation(*rotate)
+    return np.array(rotate)
+
+
+def _make_2d_rotation(theta_degrees):
+    """Makes a 2D rotation matrix from an angle in degrees."""
+    cos_theta, sin_theta = _cos_sin_degrees(theta_degrees)
+    return np.array([[cos_theta, -sin_theta], [sin_theta, cos_theta]])
 
-    # Convert a scale vector to an nD diagonal matrix
-    scale_mat = np.diag(scale)
-    n_scale = scale_mat.shape[0]
 
+def _make_3d_rotation(alpha_degrees, beta_degrees, gamma_degrees):
+    """Makes a 3D rotation matrix from roll, pitch, and yaw in degrees.
+
+    For more details, see: https://en.wikipedia.org/wiki/Rotation_matrix#General_rotations
+    """
+    cos_alpha, sin_alpha = _cos_sin_degrees(alpha_degrees)
+    R_alpha = np.array(
```

Comment:
The definition of `R_alpha` has changed to reflect the definition described in the link in the docstring: https://en.wikipedia.org/wiki/Rotation_matrix#General_rotations

Would be good for someone to double check that definition is correct, because 3D rotations are always a little tricky and I may be forgetting something.

Reply:
hmm i might have needed to have that `-` sign in there to add a flip of some form - possibly because our y axis is flipped in that positive values go down instead of up. I have tried this out though and it all works fine for me - so not sure if that `-` has now been absorbed somewhere else, or if my example wasn't complex enough to catch it. Looks like the 2D case was already like this though, so maybe my ideas here are off and it really is just a bug - either way i can't see any reason now not to make the change so am fine with it

Reply:
I think you always need the alternating signs for subsequent axes, though which side is - and which side is + might vary depending on convention. So my gut instinct is that we should go with this change. It'll become obvious quickly enough if something fishy is going on, but if it is, I suspect that the "real" fix lies elsewhere than just a one-off sign change here.